### PR TITLE
testing: stop using an old version in testServer

### PR DESCRIPTION
**Pull request by dnephin**
Our tests should run with the current version by default.

Also deprecate some functions, see commit message for the rationale.

### CODE REVIEW

File: agent/consul/server_test.go
Comment (dnephin) [inline agent/consul/server_test.go:194]: These are already set in `testServerWithConfig`, so I don't think setting them here again does anything.

File: agent/consul/server_test.go
Comment (dnephin) [inline agent/consul/server_test.go:219]: By making this a varargs, I think we'll be able to:
* rename this to `testServer` since calling it with no `configOpts` functions is fine
* instead of separate `testServerX` methods, we can create `configOpts` functions and pass them in (ex: one for enabling ACLs, one for TLS, one for secondary). That allows us to compose the setup functions without wrapping the main function.


I'd like to also remove `newServer` and rename `testServer` to `newTestServer`, but I thought I'd do all of those renames at once in a separate PR, since it will change a few hundred lines.

File: agent/consul/server_test.go
Comment (dnephin) [inline agent/consul/server_test.go:1280]: This was the only caller of `testServerDCExpectNonVoter` so I moved the settings here instead of deprecating the function.

The other ones have a few dozen callers, so I left them.

File: agent/consul/server_test.go
Comment (FFMMM) [inline agent/consul/server_test.go:167]: IIUC, there's one more instance of this in https://github.com/hashicorp/consul/blob/85fe875d0772a6da4b42c03795ab4b44e8f92182/agent/consul/leader_test.go#L1440 

these seem to be more like placeholders (gossiped around) than actual config knobs right?

https://github.com/hashicorp/consul/blob/eb21649f82bf78e27a75b012013f695d9beba6af/agent/consul/config.go#L176

**Review by FFMMM — APPROVED** [review]
📦 LGTM! Left a small nit/ ref if you want to tackle it!

File: agent/consul/server_test.go
Comment (dnephin) [reply agent/consul/server_test.go:167]: Oh ya, I think there are a few individual tests that do set this value. I think that's ok. We still want to be able to run some tests and pretend the agent is an older version. What I was fixing in this PR was defaulting to an old version. I think the default needs to be the current version.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/536234.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 1683da66b02f980834f1fa43c9744a48f4ba10a6 onto `release/1.11.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 1683da66b02f980834f1fa43c9744a48f4ba10a6 onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/536234)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 1683da66b02f980834f1fa43c9744a48f4ba10a6 onto `release/1.9.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/536234)

**Pull request merged**
